src/sentry/workflow_engine/migration_helpers/rule_action.py

```diff
@@ -0,0 +1,106 @@
+from typing import Any
+
+from sentry.notifications.models.notificationaction import ActionTarget
+from sentry.workflow_engine.models.action import Action
+from sentry.workflow_engine.typings.notification_action import (
+    ACTION_TYPE_2_INTEGRATION_ID_KEY,
+    ACTION_TYPE_2_TARGET_DISPLAY_KEY,
+    ACTION_TYPE_2_TARGET_IDENTIFIER_KEY,
+    ACTION_TYPE_2_TARGET_TYPE_RULE_REGISTRY,
+    EXCLUDED_ACTION_DATA_KEYS,
+    RULE_REGISTRY_ID_2_INTEGRATION_PROVIDER,
+    SlackDataBlob,
+)
+
+
+def build_slack_data_blob(action: dict[str, Any]) -> SlackDataBlob:
+    """
+    Builds a SlackDataBlob from the action data.
+    Only includes the keys that are not None.
+    """
+    return SlackDataBlob(
+        tags=action.get("tags", ""),
+        notes=action.get("notes", ""),
+    )
+
+
+def sanitize_to_action(action: dict[str, Any], action_type: Action.Type) -> dict[str, Any]:
+    """
+    Pops the keys we don't want to save inside the JSON field of the Action model.
+
+    :param action: action data (Rule.data.actions)
+    :param action_type: action type (Action.Type)
+    :return: action data without the excluded keys
+    """
+
+    # # If we have a specific blob type, we need to sanitize the action data to the blob type
+    if action_type == Action.Type.SLACK:
```

Comment:
i tried out a mapping for this from action_type to the dataclass, but i liked having more explicit typing here since otherwise i think i would have to put a generic type.

Reply:
open to better ideas